### PR TITLE
OPS-2175: migrating usage of request and request-promise to axios

### PR DESCRIPTION
**Pull request by aditjain117**
[OPS-2175](https://freewill.atlassian.net/browse/OPS-2175)

## How did you implement it
moving all usage of `request` and `request-promise` (and subsequently `request-progress`) to use `axios` instead. 

`request` and `request-promise` are prone to prototype pollution vulnerability from versions of `tough-cookie` older than 4.1.3 (more info in ticket)

[OPS-2175]: https://freewill.atlassian.net/browse/OPS-2175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## How can we verify it
`getReleaseSource` and `downloadFlywaySource` are called when we run `npm run install` in the flyway project (after building with our new changes of course). `getReleaseSource` should return the correct data from the Maven repository and is the first step in `npm run install` command. `downloadFlywaySource` is the second step in `npm run install` command and will save a file to your machine after success. Here are ss's of running `npm run install` locally (same output as when we used request library, and also got verbose error logging as expected when I had errors in my code!):

in the middle, we can see the progress bar working as intended:
image

when completed:
image

- test the behavior across different platforms (win32, linux, darwin, and arm64) to confirm no regression
  - how might I do this?

### CODE REVIEW

File: dist/utils/index.js [hunk 1]
```diff
@@ -17,18 +17,6 @@ var _path = require("path");
 
 var _path2 = _interopRequireDefault(_path);
 
-var _request = require("request");
```
Comment (aditjain117) [inline dist/utils/index.js:20]: ignore these `dist` changes, these are auto generated when we run `npm run build` on our src files.. scroll down to index.js in src/utils for the juicy stuff

File: package.json [hunk 2]
```diff
@@ -35,13 +35,12 @@
...
   "dependencies": {
+    "axios": "^1.7.7",
     "extract-zip": "^2.0.1",
     "filesize": "^10.1.0",
     "fs-extra": "^11.1.1",
     "progress": "^2.0.3",
-    "request": "^2.88.2",
-    "request-progress": "^3.0.0",
-    "request-promise": "^4.2.6"
+    "request-progress": "^3.0.0"
```
Comment (aditjain117) [inline package.json]: from perry - remove since not being used anymore

**Comment by aditjain117** [comment]
todo after speaking with perry:

Add validation for before changes
Add validation for running flyway command (such as dump or migrate) after connecting local changes to our main repo

**Review by perryahn — APPROVED** [review]
lgtm! can we confirm that the installed flyway can successfully run our db commands in `app`?

**Comment by aditjain117** [comment]
> lgtm! can we confirm that the installed flyway can successfully run our db commands in `app`?
@perryahn yes!! See below

locally linking flyway changes to app's package.json (after building/packing flyway changes): 
image

running npm run db-migrate:
image

running npm run db-generate:
image

**Review by perryahn — APPROVED** [review]
nice

**Review by kwalker-freewill — APPROVED** [review]

**Pull request merged**